lib/authentication/service/auth_provider.dart

```diff
@@ -184,6 +147,39 @@ class AuthProvider with ChangeNotifier {
     return _firebaseUser.uid;
   }
 
+  /// The old format of class in the data base is a Map<String, String>,
+  /// where the key is the name of the level in the filter tree.
+  /// In the new format the class is a List<String> that contains the name of
+  /// the nodes
+  Future<void> migrateToNewClassFormat(Map<String, dynamic> userData) async {
+    List classes = List();
+    classes.add(userData['class']['degree']);
+    if (userData['class']['domain'] != null) {
+      classes.add(userData['class']['domain']);
+      if (userData['class']['year'] != null) {
+        classes.add(userData['class']['year']);
+        if (userData['class']['series'] != null) {
+          classes.add(userData['class']['series']);
+          if (userData['class']['group'] != null) {
+            classes.add(userData['class']['group']);
+            if (userData['class']['subgroup'] != null) {
+              classes.add(userData['class']['subgroup']);
+            }
+          }
+        }
+      }
+    }
```

Comment:
This whole thing can be something like: `List<String> classes = ['degree', 'domain', 'year', 'series', 'group', 'subgroup'].map((key) => userData['class'][key]).where((s) => s != null).toList();`